recipes/dydxprotocol-v4-proto-cpp/meta.yaml

```
@@ -0,0 +1,180 @@
+  'dydxprotocol/subaccounts': ["subaccount.pb.h", "asset_position.pb.h", "query.pb.h", "genesis.pb.h",
+                               "perpetual_position.pb.h"],
+  'dydxprotocol/assets': ["query.pb.h", "genesis.pb.h", "tx.pb.h", "asset.pb.h"],
+  'dydxprotocol/perpetuals': ["perpetual.pb.h", "query.pb.h", "genesis.pb.h", "tx.pb.h", "params.pb.h"],
+  'dydxprotocol/feetiers': ["query.pb.h", "genesis.pb.h", "tx.pb.h", "params.pb.h"],
+  'dydxprotocol/blocktime': ["query.pb.h", "genesis.pb.h", "tx.pb.h", "blocktime.pb.h", "params.pb.h"],
+  'dydxprotocol/rewards': ["reward_share.pb.h", "query.pb.h", "genesis.pb.h", "tx.pb.h", "params.pb.h"],
+  'dydxprotocol/sending': ["query.pb.h", "genesis.pb.h", "tx.pb.h", "transfer.pb.h"],
+  'dydxprotocol/prices': ["query.pb.h", "genesis.pb.h", "tx.pb.h", "market_param.pb.h", "market_price.pb.h"],
+  'dydxprotocol/epochs': ["query.pb.h", "genesis.pb.h", "epoch_info.pb.h"],
+  'dydxprotocol/indexer/shared': ["removal_reason.pb.h"],
+  'dydxprotocol/indexer/socks': ["messages.pb.h"],
+  'dydxprotocol/indexer/redis': ["redis_order.pb.h"],
+  'dydxprotocol/indexer/events': ["events.pb.h"],
+  'dydxprotocol/indexer/off_chain_updates': ["off_chain_updates.pb.h"],
+  'dydxprotocol/indexer/indexer_manager': ["event.pb.h"],
+  'dydxprotocol/indexer/protocol/v1': ["subaccount.pb.h", "clob.pb.h"],
+  'dydxprotocol/delaymsg': ["query.pb.h", "block_message_ids.pb.h", "genesis.pb.h", "tx.pb.h", "delayed_message.pb.h"],
+  'dydxprotocol/daemons/liquidation': ["liquidation.pb.h"],
+  'dydxprotocol/daemons/pricefeed': ["price_feed.pb.h"],
+  'dydxprotocol/daemons/bridge': ["bridge.pb.h"],
+  'dydxprotocol/stats': ["query.pb.h", "genesis.pb.h", "tx.pb.h", "stats.pb.h", "params.pb.h"],
+  'dydxprotocol/bridge': ["query.pb.h", "genesis.pb.h", "tx.pb.h", "bridge_event.pb.h", "params.pb.h",
+                          "bridge_event_info.pb.h"],
+  'dydxprotocol/vest': ["query.pb.h", "genesis.pb.h", "tx.pb.h", "vest_entry.pb.h"],
+} %}
+
+{% set dydx_v4_proto_header_dir_paths = ['amino', 'cosmos', 'cosmos_proto', 'gogoproto', 'google', 'tendermint'] %}
+
+
+package:
+  name: dydx-v4-proto-cpp-split
+  version: {{ version }}
+
+source:
+  folder: all-sources
+  url: https://github.com/dydxprotocol/v4-clients/archive/refs/tags/v4-client-js@{{ version }}.tar.gz
+  sha256: eb6ffc4e53aa0a22f195092f2102fb5233c5c4dda7b11700208a1974d1428297
+  patches:
+    - patches/0001-build-proto-only.patch
+    - patches/0002-remove-unused-packages.patch
+    - patches/0003-add-sharedlib-install.patch
+    - patches/0004-remove-FetchContent.patch
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ stdlib("c") }}
+    - cmake >=3.25
+    - libprotobuf >=4.22  # [build_platform == target_platform]
+    - m2w64-binutils  # [win]
+    - ninja
+    - pkg-config
+    - zlib
+  host:
+    - catch2
+    - libabseil >=20240116
+    - libprotobuf >=4.22
```

Omit the pins to get get global pin
```suggestion
    - libabseil
    - libprotobuf
```